src/context/WalletProvider/SelectModal.tsx

```diff
@@ -100,6 +103,56 @@ const WalletSelectItem = ({
   )
 }
 
+const MipdProviderSelectItem = ({
+  provider,
+  connect,
+}: {
+  provider: EIP6963ProviderDetail
+  connect: (adapter: string) => void
+}) => {
+  const connectedRdns = useAppSelector(selectWalletRdns)
+
+  const greenColor = useColorModeValue('green.500', 'green.200')
+  const handleConnect = useCallback(
+    () => connect(provider.info.rdns),
+    [connect, provider.info.rdns],
+  )
+
+  const detectedMipdProviders = useSyncExternalStore(mipdStore.subscribe, mipdStore.getProviders)
+  const connectedMipdProvider = detectedMipdProviders.find(
+    provider => provider.info.rdns === connectedRdns,
+  )
+
+  const icon = provider.info.icon
+  const activeWallet = provider.info.name === connectedMipdProvider?.info.name
+  const walletSubText = activeWallet ? 'common.connected' : null
+
+  return (
+    <Button
+      key={provider.info.rdns}
+      w='full'
+      size='md'
+      py={8}
+      isActive={activeWallet}
+      justifyContent='space-between'
+      onClick={handleConnect}
+      data-test={`connect-wallet-${provider.info.rdns}-button`}
+    >
+      <Flex alignItems='flex-start' flexDir='column'>
+        <RawText fontWeight='semibold'>{provider.info.name}</RawText>
+        <Text fontSize='xs' color='text.subtle' translation={walletSubText} />
+      </Flex>
+      <Center width='25%'>
+        {activeWallet ? (
+          <CheckCircleIcon color={greenColor} />
+        ) : (
+          <Image width='24px' height='auto' src={icon} />
+        )}
+      </Center>
+    </Button>
+  )
```

Comment:
preferably-blocking: All this part is the exact same than `WalletSelectItem`, I would make it reusable with a neutral interface and consume it both in `WalletSelectItem` and `MipdProviderSelectItem` as only the js logic differs and not the markup

Reply:
🧠  8527362be7